### ISSUE TITLE
Exception ticking world with ICBM-Classic

### ISSUE THREAD

**Issue by Harchvertelol** [SpongePowered/SpongeForge#3143]
**I am currently running**

SpongeForge version: spongeforge-1.12.2-2838-7.2.0 (and last too)
Forge version: 14.23.5.2847
Java version: Java 8
Operating System: Windows 10

Adding sponge-forge breaks ICBM-Classic.

Link: https://github.com/BuiltBrokenModding/ICBM-Classic/issues/312

**Comment by dualspiral** [comment]
As the issue template says:

> Please provide a *full* server log (and crash-report if applicable).
> Go to https://gist.github.com/ and upload them there, then paste the resulting link here!
> Don't use hastebin/pastebin or other similar sites, as they have a history of quickly
> deleting files before we can look at them. 

Both `debug.log` and `crash-report*` files are useful here. Linking to another issue, while useful, doesn't help if that gets edited/altered.

Also please explain how it breaks, how to reproduce it, etc.

**Comment by Harchvertelol** [comment]
This is described at the specified link. If this is so important, I can copy the description from there.

**Comment by Harchvertelol** [comment]
debug.log and latest.log:

https://gist.github.com/Harchvertelol/521276a30b553f574c2f5db0ca730e4c

**Comment by Harchvertelol** [comment]
Copy from link:

I tested this for two version of ICBM: ICBM - Classic 1.12.2-4.0.1b75-UNSTABLE build 75 and ICBM-classic-1.12.2-4.0.1.

My actions:

1) Make a launcher and launch a rocket at a distance of 2000 2000 blocks diagonally (for example, from point 1000 1000 to point -1000 -1000).
2) Almost immediately, teleport to the rocket's arrival point (after 5 seconds, when the rocket is already in the sky).
3) The server crashes with an error below.

All missile flight chunks were generated in advance (suddenly this is important).

**Comment by Harchvertelol** [comment]
Important part of log:

LOG:

[00:34:42] [Server thread/ERROR] [minecraft/MinecraftServer]: Encountered an unexpected exception
net.minecraft.util.ReportedException: Exception ticking world
at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:762) ~[MinecraftServer.class:?]
at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:397) ~[nz.class:?]
at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668) ~[MinecraftServer.class:?]
at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
at java.lang.Thread.run(Unknown Source) [?:1.8.0_251]
Caused by: java.util.ConcurrentModificationException
at java.util.ArrayList$Itr.checkForComodification(Unknown Source) ~[?:1.8.0_251]
at java.util.ArrayList$Itr.next(Unknown Source) ~[?:1.8.0_251]
at icbm.classic.lib.radar.RadarMap.collectEntitiesInChunk(RadarMap.java:228) ~[RadarMap.class:1.12.2-4.0.1b20200328215453]
at icbm.classic.content.entity.missile.MissileEventHandler.chunkUnload(MissileEventHandler.java:43) ~[MissileEventHandler.class:1.12.2-4.0.1b20200328215453]
at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_22_MissileEventHandler_chunkUnload_Unload.invoke(.dynamic) ~[?:?]
at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182) ~[EventBus.class:?]
at net.minecraft.world.chunk.Chunk.func_76623_d(Chunk.java:879) ~[axw.class:?]
at net.minecraft.world.gen.ChunkProviderServer.func_73156_b(ChunkProviderServer.java:269) ~[on.class:?]
at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:207) ~[oo.class:?]
at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:756) ~[MinecraftServer.class:?]
... 4 more
[00:34:42] [Server thread/ERROR] [minecraft/MinecraftServer]: This crash report has been saved to: E:\Games\Minecraft\1.12.2\servertech.\crash-reports\crash-2020-04-28_00.34.42-server.txt
[00:34:42] [Server thread/INFO] [minecraft/MinecraftServer]: Stopping server
[00:34:42] [Server thread/INFO] [minecraft/MinecraftServer]: Saving players
[00:34:42] [Server thread/INFO] [minecraft/MinecraftServer]: Saving worlds
[00:34:42] [Server thread/INFO] [minecraft/MinecraftServer]: Saving chunks for level 'world'/overworld
[00:34:42] [Server thread/INFO] [FML]: Unloading dimension 0
[00:34:42] [Server thread/INFO] [FML]: The state engine was in incorrect state SERVER_STOPPING and forced into state SERVER_STOPPED. Errors may have been discarded.

**Comment by Harchvertelol** [comment]
Coping text about my experience:

I tested this on the server only with IC2 and ICBM-Classic (stable). It works. Adding else only a sponge breaks everything.

I made the same server without sponge - everything works. What can I do? Why does sponge-forge break it?

**Comment by clienthax** [comment]
Investigated, Problem is on ICBM's end, ICBM issue updated.
